engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/serialization/ConfluentKafkaAvroDeserializationSchemaFactory.scala

```diff
@@ -51,16 +48,20 @@ class ConfluentKafkaAvroDeserializationSchemaFactory(schemaRegistryClientFactory
 abstract class ConfluentKeyValueKafkaAvroDeserializationFactory(schemaRegistryClientFactory: ConfluentSchemaRegistryClientFactory)
   extends KafkaAvroKeyValueDeserializationSchemaFactory with ConfluentKafkaAvroDeserializerFactory {
 
-  override protected def createKeyDeserializer(kafkaConfig: KafkaConfig): Deserializer[K] =
-    createDeserializer[K](schemaRegistryClientFactory, kafkaConfig, None, isKey = true)(keyClassTag)
+  override protected def createKeyDeserializer(schemaDataOpt: Option[RuntimeSchemaData], kafkaConfig: KafkaConfig, keyClassTagOpt: Option[ClassTag[_]]): Deserializer[K] =
+    schemaDataOpt
+      .map(schema => createDeserializer[K](schemaRegistryClientFactory, kafkaConfig, Some(schema), isKey = true)(keyClassTagOpt.getOrElse(classTag[Any]).asInstanceOf[ClassTag[K]]))
+      .getOrElse(KafkaAvroKeyValueDeserializationSchemaFactory.fallbackKeyAsStringDeserializer.asInstanceOf[Deserializer[K]])
```

Comment:
It is not a good option to handle the situation when `schemaDataOpt` is `None`. We have one, not well documented feature, that we can create avro deserializer that hasn't got specified "target" schema. It wil skip only deserialize data using schema id inside message, without schema evolution. Take a look at how it is handled in `value`. our base implementation `(ConfluentKeyValueKafkaAvroDeserializationFactory)` should behave this way. If someone would like to use string as a key, he/she always can override this behaviour. WDYT?

Reply:
Fixed.